clippy_lints/src/utils/author.rs

```diff
@@ -93,11 +93,21 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Author {
     }
 
     fn check_variant(&mut self, cx: &LateContext<'a, 'tcx>, var: &'tcx hir::Variant<'_>) {
+        use rustc_hir::def_id::{LocalDefId, CRATE_DEF_INDEX};
+        use rustc_hir::{HirId, DUMMY_ITEM_LOCAL_ID};
+
+        const DUMMY_HIR_ID: HirId = HirId {
+            owner: LocalDefId {
+                local_def_index: CRATE_DEF_INDEX,
+            },
+            local_id: DUMMY_ITEM_LOCAL_ID,
+        };
+
         if !has_attr(cx.sess(), &var.attrs) {
             return;
         }
         prelude();
-        PrintVisitor::new("var").visit_variant(var, &hir::Generics::empty(), hir::DUMMY_HIR_ID);
+        PrintVisitor::new("var").visit_variant(var, &hir::Generics::empty(), DUMMY_HIR_ID);
```

Comment:
Something like `cx.tcx.hir().get_parent(var.hir_id)` should give you a valid `HirId`.

Reply:
Which probably means we can remove the `HirId` arguments from `visit_variant` and `visit_variant_data` and `visit_enum_def`, maybe. Depends on what visitors use them and for what.

Reply:
Thanks!
I'm not familiar with how the author lint works, I switched to `cx.tcx.hir().get_parent_node(var.id);` now.